File: app/src/main/java/nz/co/warehouseandroidtest/FooterViewHolder.java
package nz.co.warehouseandroidtest;

import androidx.recyclerview.widget.RecyclerView;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.TextView;

public class FooterViewHolder extends RecyclerView.ViewHolder {
    ProgressBar pbLoading;
    TextView tvLoading;
    LinearLayout llEnd;

    public FooterViewHolder(View itemView) {
        super(itemView);

        pbLoading = (ProgressBar) itemView.findViewById(R.id.pb_loading);
        tvLoading = (TextView) itemView.findViewById(R.id.tv_loading);
        llEnd = (LinearLayout) itemView.findViewById(R.id.ll_end);
    }
}

File: app/src/main/java/nz/co/warehouseandroidtest/FooterViewHolder.kt
package nz.co.warehouseandroidtest

import android.view.View
import android.widget.LinearLayout
import android.widget.ProgressBar
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class FooterViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    @JvmField
    var pbLoading: ProgressBar
    @JvmField
    var tvLoading: TextView
    @JvmField
    var llEnd: LinearLayout

    init {
        pbLoading = itemView.findViewById<View>(R.id.pb_loading) as ProgressBar
        tvLoading = itemView.findViewById<View>(R.id.tv_loading) as TextView
        llEnd = itemView.findViewById<View>(R.id.ll_end) as LinearLayout
    }
}